linkerd/app/gateway/src/lib.rs

```diff
@@ -81,7 +81,7 @@ where
 {
     let inbound_config = inbound.config().clone();
     let local_id = identity::LocalId(inbound.identity().name().clone());
-
+    let profiles = profiles::RecoverDefault::new(profiles.into_service());
```

Comment:
In the other branch I've moved this stuff up to where the profile client is built.